content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,366 @@
+
+If we were to target the input element via its `className`, the snippet above would now be:
+
+```js
+const fullnameInputField = document.querySelector(“.fullname”)
+```
+
+Note that the pound sign `(#)` has changed to the period sign `(.)` because we’re targeting that DOM node via its className property.
+
+Now let’s have a look at the validation snippet below:
+
+```js
+let errMsg;
+
+const validateSignUp = (email, password, confirmPassword, alert) => {
+  // targeting all form fields
+  const emailInput = document.querySelector("#email");
+  const passwordInput = document.querySelector("#pwd");
+  const confirmPasswordInput = document.querySelector("#pwd_conf");
+
+  if (!email) {
+    alert.error("Please input your email address");
+    emailInput.focus();
+    errMsg = false;
+  } else if (password === "") {
+    alert.error("Please provide your password!");
+    passwordInput.focus();
+    errMsg = false;
+  } else if (password.value <= 7) {
+    alert.error("Your password must have 8 characters or greater");
+    passwordInput.focus();
+    errMsg = false;
+  } else if (
+    typeof password !== "undefined" &&
+    typeof confirmPassword !== "undefined"
+  ) {
+    if (password != confirmPassword) {
+      alert.error("Passwords don't match");
+      confirmPasswordInput.focus();
+      errMsg = false;
+    }
+  } else {
+    alert.success("You've signed up successfully. Proceed to login");
+    errMsg = true;
+  }
+  return errMsg;
+};
+
+export default signUpCheck;
+```
+
+Wow! I know that might be your reaction right now, but do not fret, We’d have a breakdown of what each snippet does as we move on.
+
+- **Passing form values as props**
+  The snippet above is a helper function that performs validation on the type of data that gets into the form field before it is sent to the backend server. The function accepts the `email`, `password`, `confirmPassword`, and `alert` as arguments.
+
+```js
+let errMsg;
+
+const validateSignUp = (email, password, confirmPassword, alert) => {};
+```
+
+We’re passing `alert` as an argument to the function so that we can have access to it when it is imported into the app component. The same thing goes for the other arguments too.
+
+The `errMsg` variable serves as the container that stores and thus displays the error message from the logic. That is why we declared it with the `let` statement, so it can be reassigned to any other value in the code execution process.
+
+- **The conditional statements**
+
+  - The first condition checks if the email field is empty, if it is, an alert error message is rendered in the user interface due to the execution of `alert.error()`
+
+  The string inside the parenthesis is what will be displayed on the UI, and the browser makes sure that the input field is in focus so that it catches the attention of the user.
```

Comment:
```suggestion
  The `alert.error()` method takes a string as a parameter. The string is what will be displayed on the user interface, and the browser makes sure that the input field is in focus so that it catches the attention of the user.
```